### PR TITLE
HW10: Session, Error Checking, View Objects and Web Test

### PR DESCRIPTION
**Pull request by tienyulin**
1. Add Session and error checking
2. Add View objects
3. Add Web Tests
4. Add donation API

### CODE REVIEW

File: spec/tests/acceptance/acceptance_spec.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+require 'headless'
+require 'webdrivers/chromedriver'
+require 'watir'
+
+describe 'Acceptance Tests' do
+  before do
+    DatabaseHelper.wipe_database
+    # @headless = Headless.new
+    @browser = Watir::Browser.new
```
Comment (chinyakao) [inline spec/tests/acceptance/acceptance_spec.rb:15]: Here is a commented code

File: spec/tests/acceptance/acceptance_spec.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  end
+
+  after do
+    @browser.close
+    # @headless.destroy
+  end
```
Comment (chinyakao) [inline spec/tests/acceptance/acceptance_spec.rb:21]: commented code

**Review by chinyakao — COMMENTED** [review]
@tienyulin @StevenTimeTW @EdDuron Please take a look

**Comment by tienyulin** [comment]
@chinyakao Will fix that in the next PR, thanks a lot.

**Pull request merged**